src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -198,25 +195,33 @@ public String getScript() {
         // The temporary variable is to ensure JENKINS_SERVER_COOKIE=durable-… does not appear even in argv[], lest it be confused with the environment.
         envVars.put(cookieVariable, "please-do-not-kill-me");
 
-        List<String> launcherCmd;
-        FilePath binary = nodeRoot.child(agentInfo.getBinaryPath());
-        try (InputStream binaryStream = DurableTask.class.getResourceAsStream(binary.getName())) {
-            if (FORCE_BINARY_WRAPPER && (binaryStream != null)) {
-                FilePath controlDir = c.controlDir(ws);
-                if (!agentInfo.isBinaryCached()) {
-                    binary.copyFrom(binaryStream);
-                    binary.chmod(0755);
-                }
-                launcherCmd = binaryLauncherCmd(c, ws, shell,
-                                                controlDir.getRemote(),
-                                                binary.getRemote(),
-                                                scriptPath,
-                                                cookieValue,
-                                                cookieVariable);
+        List<String> launcherCmd = null;
+        if (agentInfo.isBinaryCompatible()) {
```

Comment:
I would move the `FORCE_BINARY_WRAPPER` check up here and delete it from the other spot so we do less work in the default case:

 ```suggestion
         if (FORCE_BINARY_WRAPPER && agentInfo.isBinaryCompatible()) {
```